### ISSUE TITLE
[Bug]: crash with NWPathMonitor 

### ISSUE THREAD

**Issue by LinKeymy** [fluttercommunity/plus_plugins#1953]
### Platform

IOS 16+

### Plugin

connectivity_plus

### Version

2.3.6+1

### Flutter SDK

3.3.5

### Steps to reproduce

crash on app launch

### Logs

```shell
Crashed: com.apple.root.background-qos
0  libsystem_kernel.dylib         0x7200 __pthread_kill + 8
1  libsystem_pthread.dylib        0x71ac pthread_kill + 268
2  libsystem_c.dylib              0x20ca0 abort + 180
3  libsystem_malloc.dylib         0x1d548 _malloc_put + 550
4  libsystem_malloc.dylib         0x1d7e0 malloc_zone_error + 100
5  libsystem_malloc.dylib         0x7df0 szone_free + 464
6  libswiftCore.dylib             0x3da1d4 _swift_release_dealloc + 56
7  libswiftNetwork.dylib          0x39f24 assignWithTake for NWPath + 116
8  libswiftNetwork.dylib          0x3a908 outlined assign with take of NWPath + 84
9  libswiftNetwork.dylib          0x3721c closure #1 in NWPathMonitor.init(requiredInterfaceType:) + 160
10 libswiftNetwork.dylib          0x2470 thunk for @escaping @callee_guaranteed (@guaranteed OS_nw_path) -> () + 52
11 Network                        0x9209d4 __nw_path_necp_update_evaluator_block_invoke + 456
12 libdispatch.dylib              0x24b4 _dispatch_call_block_and_release + 32
13 libdispatch.dylib              0x3fdc _dispatch_client_callout + 20
14 libdispatch.dylib              0x15b8c _dispatch_root_queue_drain + 684
15 libdispatch.dylib              0x16284 _dispatch_worker_thread2 + 164
16 libsystem_pthread.dylib        0xdbc _pthread_wqthread + 228
17 libsystem_pthread.dylib        0xb98 start_wqthread + 8
```

### Flutter Doctor

```shell
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.3.5, on macOS 13.0.1 22A400 darwin-arm, locale zh-Hans-CN)
[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.

[✓] Xcode - develop for iOS and macOS (Xcode 14.1)
[✓] Chrome - develop for the web
[!] Android Studio (version 2022.2)
    ✗ Unable to find bundled Java version.
[✓] IntelliJ IDEA Ultimate Edition (version 2022.3.2)
[✓] VS Code (version 1.79.2)
[✓] Connected device (4 available)
[✓] HTTP Host Availability

! Doctor found issues in 2 categories.
```

### Checklist before submitting a bug

- [X] I Google'd a solution and I couldn't find it
- [X] I searched on StackOverflow for a solution and I couldn't find it
- [X] I read the README.md file of the plugin
- [X] I'm using the latest version of the plugin
- [X] All dependencies are up to date with `flutter pub upgrade`
- [X] I did a `flutter clean`
- [X] I tried running the example project

**Comment by madilsharief** [comment]
Its strange that no one has posted solution to this problem! Its occuring in iOS 16.5.1 as well Xcode 15 Beta. App works fine in old iOS and also in iOS 17 on simulator but doesn't work in iOS 16.5.1.

Just upgraded iOS to 16.6 now and still no luck. Same error!

**Comment by vbuberen** [comment]
It is already resolved with release of `connectivity_plus: 5.0.0`
See #2169
